python/ray/tune/trial.py

```diff
@@ -330,9 +330,6 @@ def should_stop(self, result):
         if result.get(DONE):
             return True
 
-        if callable(self.stopping_criterion):
```

Comment:
i think its a little weird if this function handles stopping for dict criteria (see below) but not callables. perhaps move the rest out into a `DictStopper` for consistency and then get rid of this func entirely? 

also that way `experiment.stopper` is always a `Stopper` rather than a `dict|Stopper`

Reply:
The two considerations against this are that 1. this involves hundreds of lines to be changed across the testing codebase because for most tests, we set stopping criteria at the trial level and 2. its actually currently possible to grid search stopping criterion but this suggestion negates that.